RollupIndexerStateTests#testIndexing can fail with assertion error.

Unfortunately the failure doesn't reproduce for me locally. This failure has popped up several times in CI over the past couple days.

---

Link to the build: https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+multijob-unix-compatibility/os=debian-9/321/console

Command to reproduce: 
```
./gradlew :x-pack:plugin:rollup:test --tests "org.elasticsearch.xpack.rollup.job.RollupIndexerStateTests.testIndexing" \
  -Dtests.seed=3DEB668028ED149F \
  -Dtests.security.manager=true \
  -Dtests.locale=tg-TJ \
  -Dtests.timezone=US/Aleutian \
  -Dcompiler.java=12 \
  -Druntime.java=11
```

Relevant excerpt from the logs:
```
FAILURE 0.07s J3 | RollupIndexerStateTests.testIndexing <<< FAILURES!
  2> NOTE: All tests run in this JVM: [SearchActionTests, RollupResponseTranslationTests, RollupIndexerStateTests]
   > Throwable #1: java.lang.AssertionError: 
   > Expected: <STARTED>
   >      but: was <INDEXING>
   > 	at __randomizedtesting.SeedInfo.seed([53646ABBE2DD77C:E1EC09A2985894F7]:0)
   > 	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
   > 	at org.elasticsearch.xpack.rollup.job.RollupIndexerStateTests.testIndexing(RollupIndexerStateTests.java:281)
   > 	at java.lang.Thread.run(Thread.java:748)
```

--- a/x-pack/plugin/rollup/src/test/java/org/elasticsearch/xpack/rollup/job/RollupIndexerStateTests.java
+++ b/x-pack/plugin/rollup/src/test/java/org/elasticsearch/xpack/rollup/job/RollupIndexerStateTests.java
@@ -256,7 +256,6 @@ public class RollupIndexerStateTests extends ESTestCase {
         }
     }
 
-    @AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/issues/41046")
     public void testIndexing() throws Exception {
         RollupJob job = new RollupJob(ConfigTestHelpers.randomRollupJobConfig(random()), Collections.emptyMap());
         AtomicReference<IndexerState> state = new AtomicReference<>(IndexerState.STOPPED);
@@ -267,8 +266,8 @@ public class RollupIndexerStateTests extends ESTestCase {
                 @Override
                 protected void onFinish(ActionListener<Void> listener) {
                     super.onFinish(ActionListener.wrap(r -> {
-                        isFinished.set(true);
                         listener.onResponse(r);
+                        isFinished.set(true);
                     }, listener::onFailure));
                 }
             };
@@ -316,8 +315,8 @@ public class RollupIndexerStateTests extends ESTestCase {
                 @Override
                 protected void onFinish(ActionListener<Void> listener) {
                     super.onFinish(ActionListener.wrap(r -> {
-                        isFinished.set(true);
                         listener.onResponse(r);
+                        isFinished.set(true);
                     }, listener::onFailure));
                 }
             };

